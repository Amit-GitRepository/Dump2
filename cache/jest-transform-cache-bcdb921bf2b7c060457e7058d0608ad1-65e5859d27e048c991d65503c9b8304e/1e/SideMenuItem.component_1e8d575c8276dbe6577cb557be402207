Object.defineProperty(exports,"__esModule",{value:true});var cov_2c1re8rlcv=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/SideMenu/SideMenuItem.component.js',hash='da459df97586f3aa6e885a345aa7fda403214dda',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/SideMenu/SideMenuItem.component.js',statementMap:{'0':{start:{line:15,column:33},end:{line:17,column:3}},'1':{start:{line:16,column:4},end:{line:16,column:34}},'2':{start:{line:19,column:34},end:{line:19,column:81}},'3':{start:{line:19,column:40},end:{line:19,column:81}},'4':{start:{line:22,column:79},end:{line:22,column:89}},'5':{start:{line:23,column:4},end:{line:51,column:5}},'6':{start:{line:24,column:29},end:{line:24,column:60}},'7':{start:{line:26,column:8},end:{line:29,column:15}},'8':{start:{line:31,column:27},end:{line:31,column:98}},'9':{start:{line:32,column:6},end:{line:50,column:8}},'10':{start:{line:37,column:16},end:{line:45,column:28}},'11':{start:{line:52,column:4},end:{line:62,column:6}},'12':{start:{line:66,column:0},end:{line:69,column:2}},'13':{start:{line:71,column:0},end:{line:81,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:15,column:21},end:{line:15,column:22}},loc:{start:{line:15,column:33},end:{line:17,column:3}},line:15},'1':{name:'(anonymous_1)',decl:{start:{line:15,column:33},end:{line:15,column:34}},loc:{start:{line:15,column:39},end:{line:17,column:3}},line:15},'2':{name:'(anonymous_2)',decl:{start:{line:19,column:23},end:{line:19,column:24}},loc:{start:{line:19,column:34},end:{line:19,column:81}},line:19},'3':{name:'(anonymous_3)',decl:{start:{line:19,column:34},end:{line:19,column:35}},loc:{start:{line:19,column:40},end:{line:19,column:81}},line:19},'4':{name:'(anonymous_4)',decl:{start:{line:21,column:2},end:{line:21,column:3}},loc:{start:{line:21,column:12},end:{line:63,column:3}},line:21},'5':{name:'(anonymous_5)',decl:{start:{line:36,column:24},end:{line:36,column:25}},loc:{start:{line:37,column:16},end:{line:45,column:28}},line:37}},branchMap:{'0':{loc:{start:{line:23,column:4},end:{line:51,column:5}},type:'if',locations:[{start:{line:23,column:4},end:{line:51,column:5}},{start:{line:23,column:4},end:{line:51,column:5}}],line:23},'1':{loc:{start:{line:23,column:8},end:{line:23,column:33}},type:'binary-expr',locations:[{start:{line:23,column:8},end:{line:23,column:13}},{start:{line:23,column:17},end:{line:23,column:33}}],line:23},'2':{loc:{start:{line:31,column:27},end:{line:31,column:98}},type:'cond-expr',locations:[{start:{line:31,column:58},end:{line:31,column:91}},{start:{line:31,column:94},end:{line:31,column:98}}],line:31},'3':{loc:{start:{line:31,column:27},end:{line:31,column:55}},type:'binary-expr',locations:[{start:{line:31,column:27},end:{line:31,column:41}},{start:{line:31,column:45},end:{line:31,column:55}}],line:31},'4':{loc:{start:{line:39,column:21},end:{line:39,column:74}},type:'cond-expr',locations:[{start:{line:39,column:34},end:{line:39,column:67}},{start:{line:39,column:70},end:{line:39,column:74}}],line:39},'5':{loc:{start:{line:55,column:11},end:{line:55,column:59}},type:'cond-expr',locations:[{start:{line:55,column:19},end:{line:55,column:52}},{start:{line:55,column:55},end:{line:55,column:59}}],line:55}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0,'7':0,'8':0,'9':0,'10':0,'11':0,'12':0,'13':0},f:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0},b:{'0':[0,0],'1':[0,0],'2':[0,0],'3':[0,0],'4':[0,0],'5':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _Accordion=require('../Shared/Accordion/Accordion.component');var _Accordion2=_interopRequireDefault(_Accordion);
var _Icon=require('../Shared/Icon/Icon.component');var _Icon2=_interopRequireDefault(_Icon);
var _ISText=require('../Shared/ISText/ISText.component');var _ISText2=_interopRequireDefault(_ISText);
var _noop=require('lodash/noop');var _noop2=_interopRequireDefault(_noop);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _SideMenuItem=require('./SideMenuItem.style');var _SideMenuItem2=_interopRequireDefault(_SideMenuItem);
var _Touchable=require('../Shared/Touchable/Touchable.component');var _Touchable2=_interopRequireDefault(_Touchable);
var _constants=require('../../themes/constants.styles');
var _helper=require('../../language/i18n/helper');
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

SideMenuItem=function(_Component){_inherits(SideMenuItem,_Component);function SideMenuItem(){var _ref;var _temp,_this,_ret;_classCallCheck(this,SideMenuItem);for(var _len=arguments.length,args=Array(_len),_key=0;_key<_len;_key++){args[_key]=arguments[_key];}return _ret=(_temp=(_this=_possibleConstructorReturn(this,(_ref=SideMenuItem.__proto__||Object.getPrototypeOf(SideMenuItem)).call.apply(_ref,[this].concat(args))),_this),_this.

navigateToScreen=function(screen){cov_2c1re8rlcv.f[0]++;cov_2c1re8rlcv.s[0]++;return function(){cov_2c1re8rlcv.f[1]++;cov_2c1re8rlcv.s[1]++;
_this.props.goToScreen(screen);
};},_this.

toggleSideMenuItem=function(index){cov_2c1re8rlcv.f[2]++;cov_2c1re8rlcv.s[2]++;return function(){cov_2c1re8rlcv.f[3]++;cov_2c1re8rlcv.s[3]++;return _this.props.toggleSideMenuAccordion(index);};},_temp),_possibleConstructorReturn(_this,_ret);}_createClass(SideMenuItem,[{key:'render',value:function render()

{var _this2=this;cov_2c1re8rlcv.f[4]++;var _ref2=(cov_2c1re8rlcv.s[4]++,
this.props),icon=_ref2.icon,title=_ref2.title,screen=_ref2.screen,index=_ref2.index,nodes=_ref2.nodes,openAccordionIndex=_ref2.openAccordionIndex,lastIndex=_ref2.lastIndex;cov_2c1re8rlcv.s[5]++;
if((cov_2c1re8rlcv.b[1][0]++,nodes)&&(cov_2c1re8rlcv.b[1][1]++,nodes.length>0)){cov_2c1re8rlcv.b[0][0]++;
var collapsedState=(cov_2c1re8rlcv.s[6]++,!(index===openAccordionIndex));
var header=(cov_2c1re8rlcv.s[7]++,
_react2.default.createElement(_reactNative.View,{style:_SideMenuItem2.default.container},
_react2.default.createElement(_Icon2.default,{name:icon,size:_constants.fonts.FONT_SIZE_XL,color:_constants.colors.PRIMARY_ACTIONABLE}),
_react2.default.createElement(_ISText2.default,{type:'BOLD',style:_SideMenuItem2.default.menuTitle},(0,_helper.translate)(title))));


var headerSuffix=(cov_2c1re8rlcv.s[8]++,(cov_2c1re8rlcv.b[3][0]++,collapsedState)&&(cov_2c1re8rlcv.b[3][1]++,!lastIndex)?(cov_2c1re8rlcv.b[2][0]++,_react2.default.createElement(_reactNative.View,{style:_SideMenuItem2.default.separator})):(cov_2c1re8rlcv.b[2][1]++,null));cov_2c1re8rlcv.s[9]++;
return(
_react2.default.createElement(_Accordion2.default,{header:header,headerSuffix:headerSuffix,collapsedState:collapsedState,onToggle:this.toggleSideMenuItem(index)},
_react2.default.createElement(_reactNative.View,null,

nodes.map(function(child,childIndex)
{cov_2c1re8rlcv.f[5]++;cov_2c1re8rlcv.s[10]++;return _react2.default.createElement(_Touchable2.default,{key:childIndex,onPress:_this2.navigateToScreen(child.screen)},
_react2.default.createElement(_reactNative.View,{style:_SideMenuItem2.default.subMenuItem},
childIndex?(cov_2c1re8rlcv.b[4][0]++,_react2.default.createElement(_reactNative.View,{style:_SideMenuItem2.default.separator})):(cov_2c1re8rlcv.b[4][1]++,null),
_react2.default.createElement(_reactNative.View,{style:_SideMenuItem2.default.subMenuContent},
_react2.default.createElement(_reactNative.View,{style:_SideMenuItem2.default.listMarker}),
_react2.default.createElement(_ISText2.default,{type:'BOLD',style:_SideMenuItem2.default.subMenuTitle},(0,_helper.translate)(child.title)))));


}))));





}else{cov_2c1re8rlcv.b[0][1]++;}cov_2c1re8rlcv.s[11]++;
return(
_react2.default.createElement(_Touchable2.default,{onPress:this.navigateToScreen(screen)},
_react2.default.createElement(_reactNative.View,null,
index?(cov_2c1re8rlcv.b[5][0]++,_react2.default.createElement(_reactNative.View,{style:_SideMenuItem2.default.separator})):(cov_2c1re8rlcv.b[5][1]++,null),
_react2.default.createElement(_reactNative.View,{style:_SideMenuItem2.default.container},
_react2.default.createElement(_Icon2.default,{name:icon,size:_constants.fonts.FONT_SIZE_XL,color:_constants.colors.PRIMARY_ACTIONABLE}),
_react2.default.createElement(_ISText2.default,{type:'BOLD',style:_SideMenuItem2.default.menuTitle},(0,_helper.translate)(title))))));




}}]);return SideMenuItem;}(_react.Component);cov_2c1re8rlcv.s[12]++;


SideMenuItem.defaultProps={
goToScreen:_noop2.default,
toggleSideMenuAccordion:_noop2.default};cov_2c1re8rlcv.s[13]++;


SideMenuItem.propTypes={
goToScreen:_propTypes2.default.func,
icon:_propTypes2.default.string,
title:_propTypes2.default.string,
screen:_propTypes2.default.string,
index:_propTypes2.default.number,
nodes:_propTypes2.default.array,
toggleSideMenuAccordion:_propTypes2.default.func,
openAccordionIndex:_propTypes2.default.number,
lastIndex:_propTypes2.default.bool};exports.default=


SideMenuItem;