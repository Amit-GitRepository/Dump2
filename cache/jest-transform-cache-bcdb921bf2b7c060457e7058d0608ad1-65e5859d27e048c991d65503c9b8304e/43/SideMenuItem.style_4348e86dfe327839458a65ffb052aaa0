Object.defineProperty(exports,"__esModule",{value:true});var cov_1b1v7za8oi=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/SideMenu/SideMenuItem.style.js',hash='cdcbdb31e8f4552a90a518d41494c321a2699bf0',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/SideMenu/SideMenuItem.style.js',statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _constants=require('../../themes/constants.styles');exports.default=

{
container:{
paddingHorizontal:15,
paddingVertical:10,
flexDirection:'row',
alignItems:'center',
alignSelf:'stretch'},

menuTitle:{
color:_constants.colors.PRIMARY_BG_TEXT_CONTRAST,
fontSize:_constants.fonts.FONT_SIZE_NORMAL,
paddingHorizontal:15},

separator:{
borderWidth:0.5,
borderColor:_constants.colors.SECONDARY_BG_TEXT_CONTRAST,
marginHorizontal:15,
alignSelf:'stretch'},

subMenuItem:{
backgroundColor:_constants.colors.PRIMARY_SUBMENU},

subMenuTitle:{
color:_constants.colors.PRIMARY_BG_TEXT_CONTRAST,
fontSize:_constants.fonts.FONT_SIZE_NORMAL,
paddingVertical:10,
marginLeft:10},

subMenuContent:{
flexDirection:'row',
alignItems:'center'},

listMarker:{
backgroundColor:_constants.colors.PRIMARY_ACTIONABLE,
borderRadius:8,
padding:4,
marginLeft:55}};