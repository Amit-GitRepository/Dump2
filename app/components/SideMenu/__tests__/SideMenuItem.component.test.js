import React from 'react';
import ReactSixteenAdapter from 'enzyme/build/adapters/ReactSixteenAdapter';
import renderer from 'react-test-renderer';
import SideMenuItem from '../SideMenuItem.component';
import {configure, shallow} from 'enzyme';

configure({adapter: new ReactSixteenAdapter()});

describe('SideMenuItem component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<SideMenuItem/>);
    const component = renderer.create(<SideMenuItem />).toJSON();
    expect(component).toMatchSnapshot();
    expect(wrapper).toBeDefined();
  });
});
