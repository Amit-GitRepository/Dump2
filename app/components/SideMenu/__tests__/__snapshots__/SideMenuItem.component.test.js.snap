// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenuItem component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "flexDirection": "row",
            "paddingHorizontal": 15,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#F93E60",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "iservice",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#FFFFFF",
              "fontFamily": "SukhumvitSet-Bold",
              "fontSize": 16,
              "paddingHorizontal": 15,
            }
          }
        >
          [missing "en." translation]
        </Text>
      </View>
    </View>
  </View>
</View>
`;
